.pre-commit-config.yaml

```
@@ -0,0 +1,32 @@
+- repo: https://github.com/ambv/black
+  rev: stable
+  hooks:
+    - id: black
+      name: Format Python Code
+      language: python
+      entry: black
+      args:
+        - --safe
+        - --line-length=79
+        - --target-version=py37
+        - .
+
+- repo: https://github.com/PyCQA/flake8
+  rev: 3.7.9
+  hooks:
+    - id: flake8
+      name: Flake8 Check
+      language: python
+      entry: flake8
+      args:
+        - feed
+
+- repo: https://github.com/pycqa/pydocstyle
+  rev: master
+  hooks:
+    - id: pydocstyle
+      name: Python Documentation Style Check
+      language: python
+      entry: pydocstyle
+      args:
+        - feed
```

minor suggestion to prevent conflicts with black (particularly D202):

```suggestion
        - feed
        - --add-ignore=D401,D202
```

---

This is already included int the `setup.cfg` file.